server/credentialsAreValid.ts

```diff
@@ -1,15 +1,19 @@
 "use server";
 
-import { validLoginDetails } from "./getCredentials";
+import prisma from "@/lib/prisma";
 
 export async function credentialsAreValid(
   inputUsername: string,
   inputPassword: string
 ) {
-  return validLoginDetails.some((loginDetails, index) => {
+  const result = await prisma.user.findMany();
+  const existingCredentials = result.map((users) => {
+    return { username: users.username, password: users.password };
+  });
+  return existingCredentials.some((loginDetails, index) => {
     return (
       loginDetails.username === inputUsername &&
-      validLoginDetails[index].password === inputPassword
+      existingCredentials[index].password === inputPassword
```

Comment:
as previously discussed, we shouldn't need to look through all the users in code - we should instead be able to look for a single user that matches the username + password combination

Reply:
copied an example from https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findunique

const result = await prisma.user.findUnique({
    where: { username: inputUsername },
  }); 

this is the error:
Type '{ username: string; }' is not assignable to type 'UserWhereUniqueInput'.
  Type '{ username: string; }' is not assignable to type '{ id: string; email: string; } & { id?: string | undefined; email?: string | undefined; AND?: UserWhereInput | UserWhereInput[] | undefined; OR?: UserWhereInput[] | undefined; ... 5 more ...; posts?: PostListRelationFilter | undefined; }'.
    Type '{ username: string; }' is missing the following properties from type '{ id: string; email: string; }': id, emailts(2322)

Reply:
https://github.com/auberginearms/groceryDatabaseApp/pull/5/commits/ad77f1bfb4c717273e06893bd54ab933437b5afa